pyanaconda/ui/gui/spokes/subscription.py

```diff
@@ -1008,4 +1009,24 @@ def _check_connectivity(self):
 
     def _update_register_button_state(self):
         """Update register button state."""
-        # TODO
+        if self._registration_controls_enabled:
+            # check if credentials are sufficient for registration
+            if self.authentication_method == AuthenticationMethod.USERNAME_PASSWORD:
+                username_set = self.subscription_request.account_username
+                password_type = self.subscription_request.account_password.type
+                password_set = password_type in (SECRET_TYPE_TEXT, SECRET_TYPE_HIDDEN)
+                if username_set and password_set:
```

Comment:
I think I have already seen these checks.

Reply:
Well we have this checks:
https://github.com/rhinstaller/anaconda/blob/master/pyanaconda/ui/lib/subscription.py#L93
But that creates a new Subscription module DBus proxy on each invocation, which I don't think is something we want to here, as this check is invoked basically on every key press on the authentication. But I guess I could add an optional `subscription_request` parameter to the checking function, that would make it skip fetching the subscription request structure over DBus.

Reply:
OK, so with the helper function it really looks quite a bit nicer! :)